tracing-appender/src/rolling.rs

```diff
@@ -668,217 +619,70 @@ impl Inner {
-            index: found_index,
-        })
-    }
-
-    fn find_filename_index(&self, now: OffsetDateTime, max_bytes: u64) -> u64 {
-        macro_rules! unwrap_or_zero {
-            ($value:expr) => {
-                if let Some(value) = $value {
-                    value
-                } else {
-                    return 0;
+                let filename = entry.file_name();
+                // if the filename is not a UTF-8 string, skip it.
+                let filename = filename.to_str()?;
+                if let Some(prefix) = &self.log_filename_prefix {
+                    if !filename.starts_with(prefix) {
+                        return None;
+                    }
                 }
-            };
-        }
-
-        let read_dir = unwrap_or_zero!(fs::read_dir(&self.log_directory).ok());
 
-        let mut files: Vec<FilteredLogFile> = read_dir
-            .filter_map(|entry| {
-                let entry = entry.ok()?;
-                self.filter_log_file(&entry)
-            })
-            .collect();
-
-        files.sort_unstable_by_key(|file| file.created_at);
-
-        let most_recent_file = unwrap_or_zero!(files.last());
-
-        let most_recent_formatted_date = unwrap_or_zero!(&most_recent_file.formatted_date).clone();
-
-        let formatted_date_now = now
-            .format(&self.date_format)
-            .expect("Unable to format OffsetDateTime; this is a bug in tracing-appender");
-
-        if most_recent_formatted_date != formatted_date_now {
-            return 0;
-        }
-
-        let latest_file = files
-            .into_iter()
-            .filter(|log_file| match &log_file.formatted_date {
-                Some(formatted_date) => most_recent_formatted_date.eq(formatted_date),
-                None => false,
-            })
-            .max_by_key(|log_file| log_file.index);
-
-        let latest_file = unwrap_or_zero!(latest_file);
-
-        let index = unwrap_or_zero!(latest_file.index);
+                if let Some(suffix) = &self.log_filename_suffix {
+                    if !filename.ends_with(suffix) {
+                        return None;
+                    }
+                }
 
-        // Increase the index if the latest file was too big
-        if latest_file.metadata.len() > max_bytes {
-            index + 1
-        } else {
-            index
-        }
-    }
+                if self.log_filename_prefix.is_none()
+                    && self.log_filename_suffix.is_none()
+                    && Date::parse(filename, &self.date_format).is_err()
+                {
+                    return None;
+                }
 
-    fn prune_old_logs(&self, max_files: usize) {
-        let files = fs::read_dir(&self.log_directory).map(|dir| {
-            dir.filter_map(|entry| {
-                let entry = entry.ok()?;
-                self.filter_log_file(&entry)
+                let created = metadata.created().ok()?;
+                Some((entry, created))
```

Comment:
issue: It’s better to use the date parsed from the filename, because the created date in the metadata of the file could be different and cause strange behaviors. Example: when copying the files around (e.g.: for backups). We would start deleting old files in a random order. That’s I recommend we keep parsing the filename for the date instead.

Reply:
Got it, now I see why you made that change in this function. I'll add your code back.

Reply:
Done, I think I got it right. In the current implementation we shouldn't need the `metadata.created()` for anything right? Or would you use it as a fallback in case we can't parse the filename as a date? I don't think this case will ever happen, as we use the same date format that was used to create the log file in the first place.

Reply:
Yes, at least my code was completely ignoring files not following the format, considering them as not managed by the logger. I think that’s the appropriate behavior.